src/doc/man/generated_txt/cargo-build.txt

```diff
@@ -51,6 +51,15 @@ OPTIONS
        binary and library targets of the selected packages. Binaries are
        skipped if they have required-features that are missing.
 
+       Binary targets are automatically built if there is an integration test
+       or benchmark. This allows an integration test to execute the binary to
+       exercise and test its behavior. The CARGO_BIN_EXE_<name> environment
```

Comment:
> Binary targets are automatically built if there is an integration test or benchmark

I think the text needs to be tweaked for `build` and `rustc`.  I'm unsure if there is a way to make it work for both those and `test` / `bench`.

Re-phrased, this is saying "If integration tests exist, then binary targets will be built".  A naked `cargo build` will build those binary targets regardless.

Some options (changes **bolded**)
- "Binary targets are automatically built if there is an integration test or benchmark **being built**."
- "Binary targets are automatically built before any integration test or benchmark"

Reply:
Changed to

> Binary targets are automatically built if there is an integration test or benchmark being selected to **build**|**test**|**benchmark**

Better?

Reply:
With the `|` are you implying you would specialize this per command?  That would work.  I had assumed "being built" was sufficient since that is a shared operation between all of them but people don't always pick up on those things.

And whoops, I left off my second suggestion and forgot what it was going to be.

Reply:
Yup. The latest commit does that specialization.

https://github.com/rust-lang/cargo/pull/10726/commits/dd8841f6f3782decfe8e53480568045b5b4240af#diff-24784c171a667761383baf6af74e0bd7e9b540cf8cfbbb0b2bfea6c00e0955fa